app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -226,7 +233,72 @@ enum SectionHeaderPlacement {
 const SECTION_HEADER_PLACEMENT: SectionHeaderPlacement = SectionHeaderPlacement::Bottom;
 
 
+
+// ========================
 // === Shape Definition ===
+// ========================
+
+
+// === Selection ===
+
+/// A shape of a selection box. It is used as a mask to show only specific parts of the selection
+/// layers. See [component_group](component_group#selection) module documentation to learn how this
+/// mask is used.
+///
+/// The selection `mask` is a rectangle with rounded corners. The final shape is an intersection of
+/// the `mask` and the `area` shape that covers the entire component list panel and has the same
+/// shape. This is needed to ensure that the selection box never leaves the borders of the
+/// component list panel. Instead of using `set_position` API methods of the display object, we
+/// have a `pos` parameter instead. Similarly, we have the `selection_size` parameter for the
+/// size of the `mask`.
+///
+/// The `area` shape consists of two parts and has a variable height. The top part is a simple
+/// rectangle, while the bottom part is a rounded rectangle, to repeat the rounded corners on the
+/// bottom of the component browser. The variable height is used to create an effect of the
+/// covering the selection with the component group's header. When the header is selected,
```

Comment:
"an effect of the covering the selection with the component group's header":
- not correct grammar.
- The "effect" is not explained. What does it mean?

Reply:
The whole section is rewritten for clarity.